stan/math/opencl/prim/add_diag.hpp

```diff
@@ -31,8 +31,9 @@ inline auto add_diag(T_m&& mat, T_a&& to_add) {  // NOLINT
     check_consistent_sizes("add_diag (OpenCL)", "number of elements of to_add",
                            to_add, "diagonal", length_diag);
   }
-  diagonal(mat) = diagonal(mat) + to_add;
-  return mat;
+  matrix_cl<typename std::decay_t<T_m>::Scalar> mat_eval = mat.eval();
```

Comment:
```suggestion
  matrix_cl<value_type_t<T_m>> mat_eval = mat;
```
Just making the code look nicer.